src/model.ts

```diff
@@ -686,16 +677,14 @@ export function build<TSchema extends BaseSchema, TDefaults extends Partial<TSch
         {
           ...model.defaultOptions,
           ...options,
-        } as CollectionInsertOneOptions
+        } as InsertOneOptions
       );
 
-      if (
-        result.result.ok === 1 &&
-        result.result.n === docs.length &&
-        result.insertedCount === docs.length &&
-        result.ops.length === docs.length
-      ) {
-        return result.ops as TSchema[];
+      if (result.acknowledged && result.insertedCount === docs.length) {
+        return documents.map((doc, index) => ({
+          ...doc,
```

Comment:
So the driver no longer returns the full inserted doc, and they decided they will only return the inserted id. I am not sure if we want to start deviating from the driver, it kind of goes against the philosophy behind this library.

Reply:
One option would be to introduce `model.insertOneLegacy` to give the consumer of the this library an escape hatch from a large refactor with a method rename. But keep `model.insertOne` in-sync with the driver.

Reply:
I'd prefer we just follow the driver. With the strong TS types consumers shouldn't have a problem identifying where they need to update their code.

Reply:
Gotta say I disagree here. I don't know why we wouldn't support a very useful feature anymore simply b/c the driver dropped support for it. This is just something that all consumers of this library will now have to implement themselves. 🤷

Reply:
I don't hate the idea of a convenience method that mimics the old functionality. Much like `findById`, you could just call it something like `insertAndFindOne` and have it do both when you need it.

On the other hand, just following the driver methods seems fine to me too. Seems like it will be an uphill battle trying to maintain legacy functionality moving forward. Would be easy enough to just do an `insertOne` and then a `findOne` if you need that functionality.

Reply:
To be clear, I'm not advocating `insertOne` + `findOne`, I'm just arguing that what is currently in the PR is good (return the `documents` here)

Reply:
But the Papr `insertOne` and `insertMany` never followed the driver return type in the first place. We intentionally returned the full document(s). The return type of the driver implementation is different.